vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailFragment.kt

```diff
@@ -693,7 +695,20 @@ class RoomDetailFragment @Inject constructor(
     }
 
     private fun setupVoiceMessageView() {
-        voiceMessagePlaybackTracker.track(VoiceMessagePlaybackTracker.RECORDING_ID, views.voiceMessageRecorderView)
+        voiceMessagePlaybackTracker.track(VoiceMessagePlaybackTracker.RECORDING_ID, object : VoiceMessagePlaybackTracker.Listener {
+            var lastKnownState: VoiceMessagePlaybackTracker.Listener.State? = null
+            override fun onUpdate(state: VoiceMessagePlaybackTracker.Listener.State) {
+                views.voiceMessageRecorderView.onUpdate(state)
+                if (lastKnownState == state) return
+                when (state) {
+                    is VoiceMessagePlaybackTracker.Listener.State.Playing,
```

Comment:
Is it also mandatory to keep the screen on for the playback? I guess this can be improve if in the future we implement a BG service.

Reply:
not mandatory but my thinking is if a voice recording is longer than 30/60 seconds~ (whatever the default screen timeout is) the user may not interact with the screen which would cause the playback to pause

I've had similar issues in the past when reading long messages